velox/connectors/hive/iceberg/tests/IcebergReadTest.cpp

```diff
@@ -831,11 +826,146 @@ TEST_F(HiveIcebergTest, testPartitionedRead) {
     splits.insert(splits.end(), icebergSplits.begin(), icebergSplits.end());
   }
 
-  HiveConnectorTestBase::assertQuery(plan, splits, "SELECT 0, '2018-04-06'");
+  assertQuery(plan, splits, "SELECT 0, '2018-04-06'");
+}
+
+TEST_F(HiveIcebergTest, schemaEvolutionRemoveColumn) {
+  auto oldRowType = ROW({"c0", "c1", "c2"}, {BIGINT(), INTEGER(), VARCHAR()});
+  auto newRowType = ROW({"c0", "c2"}, {BIGINT(), VARCHAR()});
+
+  // Write data file with old schema (c0, c1, c2).
+  std::vector<RowVectorPtr> dataVectors;
+  auto c0 = makeFlatVector<int64_t>({1, 2, 3, 4, 5});
+  auto c1 = makeFlatVector<int32_t>({10, 20, 30, 40, 50});
+  auto c2 = makeFlatVector<std::string>({"a", "b", "c", "d", "e"});
+  dataVectors.push_back(makeRowVector({c0, c1, c2}));
+  auto dataFilePath = TempFilePath::create();
+  writeToFile(dataFilePath->getPath(), dataVectors);
+
+  auto icebergSplits = makeIcebergSplits(dataFilePath->getPath());
+
+  // Expected result: c0 and c2 have values, c1 is not present.
+  std::vector<RowVectorPtr> expectedVectors;
+  expectedVectors.push_back(makeRowVector(newRowType->names(), {c0, c2}));
+
+  // Read with new schema (c0 and c2 only, c1 removed).
+  auto plan = PlanBuilder().tableScan(newRowType).planNode();
+  AssertQueryBuilder(plan).splits(icebergSplits).assertResults(expectedVectors);
+}
+
+TEST_F(HiveIcebergTest, schemaEvolutionAddColumns) {
+  auto oldRowType = ROW({"c0"}, {BIGINT()});
+  auto newRowType = ROW({"c0", "c1", "c2"}, {BIGINT(), INTEGER(), VARCHAR()});
+
+  // Write data file with old schema (only c0).
+  std::vector<RowVectorPtr> dataVectors;
+  auto c0 = makeFlatVector<int64_t>({100, 200, 300});
+  dataVectors.push_back(makeRowVector({c0}));
+  auto dataFilePath = TempFilePath::create();
+  writeToFile(dataFilePath->getPath(), dataVectors);
+  auto icebergSplits = makeIcebergSplits(dataFilePath->getPath());
+
+  // Expected result: c0 has values, c1 and c2 are NULL.
+  std::vector<RowVectorPtr> expectedVectors;
+  expectedVectors.push_back(makeRowVector(
+      {c0,
+       makeNullConstant(TypeKind::INTEGER, 3),
+       makeNullConstant(TypeKind::VARCHAR, 3)}));
```

Comment:
add a comma after the last vector to improve readability

apply the same refactoring to other calls to makeRowVector